CONTRIBUTING.md

```diff
@@ -0,0 +1,130 @@
+# Contributing to Hive
+
+Thank you for your interest in contributing to OpenShift Hive! We welcome contributions from the community.
+
+## Project Overview
+
+OpenShift Hive is an operator which runs as a service on top of Kubernetes/OpenShift.
+The Hive service can be used to provision and perform initial configuration of OpenShift clusters.
+
+Hive uses the [OpenShift installer](https://github.com/openshift/installer) for cluster provisioning.
+
+For detailed design overview and usage, please refer to the [README.md](README.md) and [documentation](./docs/).
+
+## Getting Started
+
+### Prerequisites
+
+- Go (version specified in `go.mod`)
+- Make
+
+### Development Environment
+
+You can build the project binaries using the provided `make` targets.
+
+```bash
+# Update generated code
+make update
+
+# Compile the project binaries
+make build
+
+# Clean up build artifacts
+make clean
+```
+
+See the [Developing Hive](./docs/developing.md) guide for detailed setup instructions.
+
+## Testing
+
+Before submitting a Pull Request, ensure that all tests pass and the code is verified.
+
+```bash
+# Verify generated code and formatting
+make verify
+
+# Run unit tests (excludes e2e tests)
+make test
+```
+
```

Comment:
The dependency management `make` commands should have their own section here too (`make vendor`, `make modcheck`, and `make modfix`)